src/NotepadNext/DockedEditor.cpp

```diff
@@ -154,6 +158,9 @@ void DockedEditor::addEditor(ScintillaNext *editor)
     // Create the dock widget for the editor
     ads::CDockWidget *dockWidget = new ads::CDockWidget(editor->getName());
 
+    // Make tab fixed size so that tab's title/text will not be elided
+    dockWidget->tabWidget()->setSizePolicy(QSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed));
```

Comment:
Maybe try `CDockWidgetTab::setElideMode()`? That might remove the need to set `DockAreaDynamicTabsMenuButtonVisibility` to `false` above.

https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/blob/master/src/DockWidgetTab.h#L160

Reply:
Tested, unfortunately it doesn't work.

setElide(Qt::ElideNone) has the same effect as set fixed size policy, so use elide mode to disable file name elide. However, it set elide mode will not make dynamic menu bar woks - menu bar still not shown when more files/tabs opened than view tab. 

So make menu bar can be used to select files (when more files opened than view port), dynamic menu bar have to turn off. Possibly this is a bug of ads.